.github/workflows/repackage.yml

```diff
@@ -0,0 +1,193 @@
+name: Repackage Bundles
+
+on: workflow_call
+
+jobs:
+  split-tasks:
+    runs-on: ubuntu-latest
+    env:
+      BUILD_DATA_DIR: .build-data
+    outputs:
+      parallel-jobs-number: ${{ steps.split-packages.outputs.parallel-jobs-number }}
+      parallel-array: ${{ steps.split-packages.outputs.parallel-array }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: google-github-actions/auth@v1
+        with:
+          credentials_json: '${{ secrets.GOOGLE_CREDENTIALS_KERNEL_CACHE }}'
+
+      - uses: google-github-actions/setup-gcloud@v1
+      
+      - uses: ./.github/actions/env
+
+      - name: Prepare Cache
+        run: |
+          mkdir -p "${BUILD_DATA_DIR}/cache"
+          IFS=',' read -r -a bucket_names <<< "${KERNEL_BUNDLE_BUCKET}"
+          for bucket in "${bucket_names[@]}"; do
+            if [ ! -f "${BUILD_DATA_DIR}/cache/cache.yml" ]; then
+              gsutil cp "${bucket}/cache.yml" "${BUILD_DATA_DIR}/cache/cache.yml" || true
+            fi
+          done
+          touch "${BUILD_DATA_DIR}/cache/cache.yml"
+
+      - name: Restore manifest
+        uses: actions/download-artifact@v3
+        with:
+          # this is archived by the crawl workflow
+          # see .github/workflows/crawl.yml for details
+          name: kernel-crawler-manifest
+
+      - name: List packages
+        run: make list-files
+
+      - name: Split package file
+        id: split-packages
+        run: |
+          import math
+          import os
+          import json
+
+          packages = []
+          with open('.build-data/packages.txt') as pkg:
+            packages = list(map(str.strip, pkg.readlines()))
+
+          parallel_jobs = math.ceil(len(packages) / 500)
+          if parallel_jobs > 32:
+            parallel_jobs = 32
+
+          # split the packages into chunks
+          jobs = [packages[i::parallel_jobs] for i in range(parallel_jobs)]
```

Comment:
Looks like this variable is not used for anything else except to create the `combined` map, maybe then create `combined` right away with `{i: packages[i::parallel_jobs] for i in range(parallel_jobs)}`?

Also, why exactly this order? It means that neighboring jobs are going to be executed in parallel, what could be cumbersome -- if there is something wrong with one group of jobs related to one disto, it's going to be distributed everywhere.